Fixes Maven dependencies in modules: common-jpa-impl, common, and cover-image-impl

mheyen opened the pull request:
This patch is part of the Maven-dependency-plugin patch to define
the dependencies of all Opencast modules.
It is fixing the dependencies of the following modules:
- common-jpa-impl
- common
- cover-image-impl

Related to #1893


### Your pull request should…

* [ ] have a concise title
* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [ ] be against the correct branch (features can only go into develop)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated tests
* [ ] have a clean commit history
* [ ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

mheyen committed:
Fixes Maven dependencies in modules: common-jpa-impl, common, and cover-image-impl

This patch is part of the Maven-dependency-plugin patch to define
the dependencies of all Opencast modules.
It is fixing the dependencies of the following modules:
- common-jpa-impl
- common
- cover-image-impl

Related to opencast#1893

mheyen committed:
update common/pom.xml

re-add dependency org.mnode.ical4j because the module scheduler-api seems to fail otherwise.

gregorydlogan commented:
@mheyen I *think* these are the same for `r/9.x`.  If so, can we point this there?  I'm also dealing with #1688 which makes similar changes to the same bundles.

gregorydlogan commented:
I'm seeing this on JDK 11:

```
2020-11-27T12:14:19,826 | ERROR | (AbstractJobProducer$JobRunner:343) - Error handling operation 'Generate':
java.lang.NoClassDefFoundError: org/w3c/dom/svg/SVGDocument
	at java.lang.ClassLoader.defineClass1(Native Method) ~[?:?]
	at java.lang.ClassLoader.defineClass(ClassLoader.java:1017) ~[?:?]
	at org.apache.felix.framework.BundleWiringImpl$BundleClassLoader.defineClass(BundleWiringImpl.java:2410) ~[?:?]
	at org.apache.felix.framework.BundleWiringImpl$BundleClassLoader.findClass(BundleWiringImpl.java:2194) ~[?:?]
	at org.apache.felix.framework.BundleWiringImpl.findClassOrResourceByDelegation(BundleWiringImpl.java:1607) ~[?:?]
	at org.apache.felix.framework.BundleWiringImpl.access$200(BundleWiringImpl.java:80) ~[?:?]
	at org.apache.felix.framework.BundleWiringImpl$BundleClassLoader.loadClass(BundleWiringImpl.java:2053) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]
	at org.apache.batik.dom.svg.SVGDOMImplementation.createDocument(SVGDOMImplementation.java:149) ~[?:?]
	at org.apache.batik.dom.util.SAXDocumentFactory.startElement(SAXDocumentFactory.java:615) ~[?:?]
	at org.apache.xerces.parsers.AbstractSAXParser.startElement(Unknown Source) ~[?:?]
	at org.apache.xerces.impl.XMLNSDocumentScannerImpl.scanStartElement(Unknown Source) ~[?:?]
	at org.apache.xerces.impl.XMLNSDocumentScannerImpl$NSContentDispatcher.scanRootElementHook(Unknown Source) ~[?:?]
	at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl$FragmentContentDispatcher.dispatch(Unknown Source) ~[?:?]
	at org.apache.xerces.impl.XMLDocumentFragmentScannerImpl.scanDocument(Unknown Source) ~[?:?]
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source) ~[?:?]
	at org.apache.xerces.parsers.XML11Configuration.parse(Unknown Source) ~[?:?]
	at org.apache.xerces.parsers.XMLParser.parse(Unknown Source) ~[?:?]
	at org.apache.xerces.parsers.AbstractSAXParser.parse(Unknown Source) ~[?:?]
	at org.apache.xerces.jaxp.SAXParserImpl$JAXPSAXParser.parse(Unknown Source) ~[?:?]
	at org.apache.batik.dom.util.SAXDocumentFactory.createDocument(SAXDocumentFactory.java:431) ~[?:?]
	at org.apache.batik.dom.util.SAXDocumentFactory.createDocument(SAXDocumentFactory.java:349) ~[?:?]
	at org.apache.batik.dom.svg.SAXSVGDocumentFactory.createDocument(SAXSVGDocumentFactory.java:200) ~[?:?]
	at org.apache.batik.dom.svg.SAXSVGDocumentFactory.createDocument(SAXSVGDocumentFactory.java:281) ~[?:?]
	at org.apache.batik.transcoder.XMLAbstractTranscoder.transcode(XMLAbstractTranscoder.java:129) ~[?:?]
	at org.apache.batik.transcoder.SVGAbstractTranscoder.transcode(SVGAbstractTranscoder.java:156) ~[?:?]
	at org.apache.batik.apps.rasterizer.SVGConverter.transcode(SVGConverter.java:1001) ~[?:?]
	at org.apache.batik.apps.rasterizer.SVGConverter.execute(SVGConverter.java:717) ~[?:?]
	at org.opencastproject.coverimage.impl.AbstractCoverImageService.rasterizeSvg(AbstractCoverImageService.java:279) ~[?:?]
	at org.opencastproject.coverimage.impl.AbstractCoverImageService.generateCoverImageInternal(AbstractCoverImageService.java:175) ~[?:?]
	at org.opencastproject.coverimage.impl.AbstractCoverImageService.process(AbstractCoverImageService.java:128) ~[?:?]
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:313) [!/:?]
	at org.opencastproject.job.api.AbstractJobProducer$JobRunner.call(AbstractJobProducer.java:272) [!/:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.ClassNotFoundException: org.w3c.dom.svg.SVGDocument not found by org.apache.servicemix.bundles.batik [300]
	at org.apache.felix.framework.BundleWiringImpl.findClassOrResourceByDelegation(BundleWiringImpl.java:1639) ~[?:?]
	at org.apache.felix.framework.BundleWiringImpl.access$200(BundleWiringImpl.java:80) ~[?:?]
	at org.apache.felix.framework.BundleWiringImpl$BundleClassLoader.loadClass(BundleWiringImpl.java:2053) ~[?:?]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]
	... 37 more
2020-11-27T12:14:24,475 | ERROR | (WorkflowOperationWorker:143) - Workflow operation 'operation:'cover-image', position:44, state:'FAILED'' failed
org.opencastproject.workflow.api.WorkflowOperationException: 'Cover image' job did not successfuly end
	at org.opencastproject.workflow.handler.coverimage.CoverImageWorkflowOperationHandlerBase.start(CoverImageWorkflowOperationHandlerBase.java:154) ~[?:?]
	at org.opencastproject.workflow.impl.WorkflowOperationWorker.start(WorkflowOperationWorker.java:193) ~[!/:?]
	at org.opencastproject.workflow.impl.WorkflowOperationWorker.execute(WorkflowOperationWorker.java:116) [!/:?]
	at org.opencastproject.workflow.impl.WorkflowServiceImpl.runWorkflowOperation(WorkflowServiceImpl.java:801) [!/:?]
	at org.opencastproject.workflow.impl.WorkflowServiceImpl.process(WorkflowServiceImpl.java:1833) [!/:?]
	at org.opencastproject.workflow.impl.WorkflowServiceImpl$JobRunner.call(WorkflowServiceImpl.java:2283) [!/:?]
	at org.opencastproject.workflow.impl.WorkflowServiceImpl$JobRunner.call(WorkflowServiceImpl.java:2249) [!/:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
```

mheyen committed:
Merge branch 'develop' into fix-1893-common+cover

lkiesow commented on modules/cover-image-impl/pom.xml:
```diff
@@ -33,113 +33,61 @@
-      <artifactId>batik-script</artifactId>
-      <version>${batik.version}</version>
-    </dependency>
-    <dependency>
-      <groupId>org.apache.xmlgraphics</groupId>
-      <artifactId>batik-svggen</artifactId>
-      <version>${batik.version}</version>
-    </dependency>
```
I can confirm the issue @gregorydlogan reported when executing the operation `cover-image` (just run the default workflow). My guess would be that as least this dependency, possibly more batik and Apache XML dependencies are missing.

![opencast-failed-cover-image](https://user-images.githubusercontent.com/1008395/100872399-0baa7980-34a2-11eb-9c9b-d9357c58709d.png)

mheyen committed:
Update cover-image-impl/pom.xml

fixed operation "cover-image".
Possibly more batik or other dependencies could be removed.
